.github/workflows/coverage-pr-base.yml

```diff
@@ -0,0 +1,40 @@
+# This runs as a separate job because it needs to run on the `pull_request_target` event
+# in order to access the CODECOV_TOKEN secret.
+#
+# This is safe because this doesn't run arbitrary code from PRs.
+
+name: Set Codecov PR base
+on:
+  # See safety note / doc at the top of this file.
+  pull_request_target:
```

Comment:
cc @alex, I believe this use of `pull_request_target` to be safe (unlike my last time 🙈), but if you disagree, please ping and I'll revert!

Reply:
As long as this doesn't run code from the PR, and only runs code from main/this .yml, it's fine. That looks true here.

Reply:
Yep that was my understanding too 👍